### PR TITLE
Add library file for NATS Streaming

### PR DESCRIPTION
**Pull request by kozlovic**
This is for [NATS Streaming Server](http://nats.io/documentation/streaming/nats-streaming-intro/), a streaming server of the [NATS](http://nats.io) family.

# Checklist for Review

**NOTE:** This checklist is intended for the use of the Official Images maintainers both to track the status of your PR and to help inform you and others of where we're at. As such, please leave the "checking" of items to the repository maintainers. If there is a point below for which you would like to provide additional information or note completion, please do so by commenting on the PR. Thanks! (and thanks for staying patient with us :heart:)

-	[x] associated with or contacted upstream?
  * https://github.com/nats-io/nats-streaming-server/issues/137
-	[x] does it fit into one of the common categories? ("service", "language stack", "base distribution")
-	[ ] is it reasonably popular, or does it solve a particular use case well?
-	[x] does a [documentation](https://github.com/docker-library/docs/blob/master/README.md) PR exist? (should be reviewed and merged at roughly the same time so that we don't have an empty 
image page on the Hub for very long)
  * https://github.com/docker-library/docs/pull/657
-	[x] dockerization review for best practices and cache gotchas/improvements (ala [the official review guidelines](https://github.com/docker-library/official-images/blob/master/README.md#review-guidelines))?
-	[x] 2+ dockerization review?
-	[x] ~~existing official images have been considered as a base? (ie, if `foobar` needs Node.js, has `FROM node:...` instead of grabbing `node` via other means been considered?)~~
-	[x] ~~if `FROM scratch`, tarballs only exist in a single commit within the associated history?~~
-	[x] passes current tests? any simple new tests that might be appropriate to add? (https://github.com/docker-library/official-images/tree/master/test)

### CODE REVIEW

**Comment by kozlovic** [comment]
I am sure you are all very busy, so just checking that I did not miss some steps here in the submit of this PR. If I did, please let me know. Thanks!

**Comment by chrrrles** [comment]
Heya - any updates on this?  It would be nice to have an "official" image of the nats streaming server.  Thanks all.

**Comment by awalterschulze** [comment]
Any update on this?

**Comment by derekcollison** [comment]
Folks at @docker-library-bot etc. Any updates on this?

/cc @psftw

**Comment by bjflanne** [comment]
I've not been able to connect in IRC with you @tianon any updates on this one?

**Comment by tianon** [comment]
Sorry for the long tail, as usual!

As an initial high level question, I'm looking at the documentation for this and it's described as embedding and extending the core NATS server -- would it make sense for this to be provided via tags of the existing `nats` image, or is it really a completely separate tool?

**Comment by derekcollison** [comment]
Completely separate project that uses NATS, but is a streaming and persistent messaging service. Which is exact opposite of NATS, hence the internal code name which was STAN. ;)

**Comment by tianon** [comment]
@derekcollison awesome, thanks for clarifying!  Love the codename :smile:

Will look deeper and report back!

**Comment by tianon** [comment]
Would you mind updating `test/config.sh` so that the tests which expect a fuller image than this get skipped? (https://github.com/docker-library/official-images/blob/67dcab4fc53871441cd357a253ac8385ab9794b3/test/config.sh#L175-L189 -- `nats` is an obviously good example there :smile:)

**Comment by tianon** [comment]
(The Dockerization is pretty much exactly what I'd expected given `nats` -- simple `FROM scratch` / `COPY static-binary` / `ENTRYPOINT`)

**Comment by kozlovic** [comment]
@tianon I updated `test/config.sh`, not sure if this is what you were asking. From blame, it seems that you are the one who did it for `nats` so I did not know that I would have to do that for `nats-streaming`.
Let me know if this is ok. Thanks!

File: test/config.sh [hunk 1]
```diff
@@ -174,11 +174,13 @@ globalExcludeTests+=(
 	# single-binary images
 	[hello-world_utc]=1
 	[nats_utc]=1
+        [nats-streaming_utc]=1
 	[swarm_utc]=1
 	[traefik_utc]=1
 
 	[hello-world_no-hard-coded-passwords]=1
 	[nats_no-hard-coded-passwords]=1
+        [nats-streaming_no-hard-coded-passwords]=1
```
Comment (tianon) [inline test/config.sh]: These look good (although the indentation should be tabs, not spaces), and there's one more further down (`nats_override-cmd` needs to be copied for `nats-streaming_override-cmd`). :+1:

Comment (kozlovic) [inline test/config.sh]: Ok done. I looked for all references to `nats` in the file and I think that these are the only 3 ;-)
Hope it is good now. Thanks again for your patience!

**Comment by yosifkit** [comment]
Build test of #2036; c26206b (`nats-streaming`):

```console

$ bashbrew build nats-streaming:0.2.2
Using bashbrew/cache:ec2d0ee9838c651c4e7e36e474031b48230bd33021424e2d4fde76e239f7a871 (nats-streaming:0.2.2)
Tagging nats-streaming:0.2.2
Tagging nats-streaming:latest

$ test/run.sh nats-streaming:0.2.2
testing nats-streaming:0.2.2
	'cve-2014--shellshock' [1/1]...passed

```

**Comment by yosifkit** [comment]
```diff
diff --git a/nats-streaming_latest/Dockerfile b/nats-streaming_latest/Dockerfile
new file mode 100755
index 0000000..4401994
--- /dev/null
+++ b/nats-streaming_latest/Dockerfile
@@ -0,0 +1,10 @@
+FROM scratch
+
+COPY nats-streaming-server /nats-streaming-server
+
+# Expose client and management ports
+EXPOSE 4222 8222
+
+# Run with default memory based store 
+ENTRYPOINT ["/nats-streaming-server"]
+CMD []
diff --git a/nats-streaming_latest/nats-streaming-server b/nats-streaming_latest/nats-streaming-server
new file mode 100755
index 0000000..e713f53
Binary files /dev/null and b/nats-streaming_latest/nats-streaming-server differ
```

**Comment by VojtechVitek** [comment]
Thank you!

**Pull request merged**